src/core/container.jl

```diff
@@ -167,18 +167,16 @@ increase_logevidence(pc :: ParticleContainer, logw :: Float64) =
 function resample!(
     pc :: ParticleContainer,
     randcat :: Function = Turing.Inference.resample_systematic,
-    ref :: Union{Particle, Nothing} = nothing
+    ref :: Union{Particle, Nothing} = nothing;
+    weights = weights(pc)
```

Comment:
This overloads the function `weights` by a variable. It's better to rename one of them.

Reply:
Do you have any suggestions for new names? I thought `weights` is a quite self-explanatory name for both the function and the variable, and the function could still be accessed by `Turing.Inference.weights`.

Reply:
Do you think `getweights` is more informative as the function name?

Reply:
IMO it's not more informative, personally I think calling it just `weights` indicates a bit more that the weights have to be computed first. But it seems to be in line with some other function names in Turing, such as `getlogp`, so maybe that's the best solution :slightly_smiling_face: